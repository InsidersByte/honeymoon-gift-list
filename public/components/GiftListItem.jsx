/* @flow */

import React from 'react';
import { IconButton } from 'material-ui';
import Edit from 'material-ui/svg-icons/image/edit';
import Delete from 'material-ui/svg-icons/action/delete';
import css from './GiftListItem.styl';

type PropsType = {
    gift: {
        id: number,
        name: string,
        imageUrl: string,
        requested: number,
        remaining: number,
        price: number,
    },
    onSelect: Function,
    onDelete: Function,
};

export default class GiftListItem extends React.Component {
    props: PropsType;

    onSelect = () => {
        this.props.onSelect(this.props.gift);
    };

    onDelete = () => {
        this.props.onDelete(this.props.gift);
    };

    render() {
        const { gift: { name, imageUrl, requested, remaining, price } } = this.props;

        return (
            <div className={css.root}>
                <img className={css.avatar} src={imageUrl} alt={name} />

                <div className={css.textContainer}>
                    <h3 className={css.name}>{name}</h3>
                    <h4 className={css.title}>Price: Â£{price}</h4>
                    <h4 className={css.title}>Requested: {requested}</h4>
                    <h4 className={css.title}>Remaining: {remaining}</h4>
                </div>

                <IconButton touch onClick={this.onSelect}>
                    <Edit />
                </IconButton>

                <IconButton touch onClick={this.onDelete}>
                    <Delete />
                </IconButton>
            </div>
        );
    }
}
