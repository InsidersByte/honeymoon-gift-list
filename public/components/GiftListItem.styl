$avatar-width = 60px
$top-bottom-margin = 12px

.root
    align-items stretch
    cursor move
    display flex
    padding 0 16px

    // to compete with bootstrap 'Jumbotron p'
    .description
        font-size 14px

.avatar
    border-radius 50%
    height $avatar-width
    margin $top-bottom-margin 16px $top-bottom-margin 0
    width $avatar-width

.textContainer
    flex 1
    margin-right 16px

.name
    font-size 16px
    margin $top-bottom-margin 0

.title
    font-size 14px

.actionContainer
    align-items center
    display flex

    button:last-child
        margin-left 6px
