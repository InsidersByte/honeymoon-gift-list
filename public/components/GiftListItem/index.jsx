/* @flow */

import React from 'react';
import { IconButton } from 'material-ui';
import Edit from 'material-ui/svg-icons/image/edit';
import Delete from 'material-ui/svg-icons/action/delete';
import css from './index.styl';

type PropsType = {
    gift: {
        id: number,
        name: string,
        imageUrl: string,
        requested: number,
        remaining: number,
        price: number,
    },
    onSelect: Function,
    onDelete: Function,
};

const GiftListItem = ({ gift, onSelect, onDelete }: PropsType) => {
    const { name, imageUrl, requested, remaining, price } = gift;
    const handleSelect = () => onSelect(gift);
    const handleDelete = () => onDelete(gift);

    return (
        <div className={css.root}>
            <img className={css.avatar} src={imageUrl} alt={name} />

            <div className={css.textContainer}>
                <h3 className={css.name}>{name}</h3>
                <h4 className={css.title}>Price: Â£{price}</h4>
                <h4 className={css.title}>Requested: {requested}</h4>
                <h4 className={css.title}>Remaining: {remaining}</h4>
            </div>

            <IconButton touch onClick={handleSelect}>
                <Edit />
            </IconButton>

            <IconButton touch onClick={handleDelete}>
                <Delete />
            </IconButton>
        </div>
    );
};

export default GiftListItem;
