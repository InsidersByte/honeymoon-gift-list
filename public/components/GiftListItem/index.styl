$avatar-width = 60px
$top-bottom-margin = 12px

.root
    align-items center
    cursor move
    display flex
    padding 0 16px

.avatar
    border-radius 50%
    height $avatar-width
    margin $top-bottom-margin 16px $top-bottom-margin 0
    width $avatar-width

.textContainer
    flex 1
    margin-right 16px

.name
    font-size 1.2em
    margin $top-bottom-margin 0

.title
    font-size 1em
