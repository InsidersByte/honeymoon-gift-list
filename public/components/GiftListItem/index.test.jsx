import React from 'react';
import { shallow } from 'enzyme';
import Edit from 'material-ui/svg-icons/image/edit';
import Delete from 'material-ui/svg-icons/action/delete';
import GiftListItem from './';

describe('GiftListItem', () => {
  it('should render correctly', () => {
    const gift = { name: 'gift', imageUrl: 'http://image.png', requested: 5, remaining: 10, price: 20 };
    const onSelect = jest.fn();
    const onDelete = jest.fn();

    const wrapper = shallow(<GiftListItem gift={gift} onSelect={onSelect} onDelete={onDelete} />);

    expect(wrapper).toMatchSnapshot();
    expect(onSelect).not.toHaveBeenCalled();
    expect(onDelete).not.toHaveBeenCalled();
  });

  it('should call on select when edit is clicked', () => {
    const gift = { name: 'gift', imageUrl: 'http://image.png', requested: 5, remaining: 10, price: 20 };
    const onSelect = jest.fn();
    const onDelete = jest.fn();

    const wrapper = shallow(<GiftListItem gift={gift} onSelect={onSelect} onDelete={onDelete} />);

    expect(wrapper.find(Edit).length).toBe(1);
    wrapper.find(Edit).parent().simulate('click');

    expect(onSelect).toHaveBeenCalledWith(gift);
    expect(onDelete).not.toHaveBeenCalled();
  });

  it('should call on delete when delete is clicked', () => {
    const gift = { name: 'gift', imageUrl: 'http://image.png', requested: 5, remaining: 10, price: 20 };
    const onSelect = jest.fn();
    const onDelete = jest.fn();

    const wrapper = shallow(<GiftListItem gift={gift} onSelect={onSelect} onDelete={onDelete} />);

    expect(wrapper.find(Delete).length).toBe(1);
    wrapper.find(Delete).parent().simulate('click');

    expect(onSelect).not.toHaveBeenCalled();
    expect(onDelete).toHaveBeenCalledWith(gift);
  });
});
