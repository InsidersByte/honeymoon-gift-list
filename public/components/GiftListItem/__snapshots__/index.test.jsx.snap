// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftListItem should render correctly 1`] = `
<div
  className="root"
>
  <img
    alt="gift"
    className="avatar"
    src="http://image.png"
  />
  <div
    className="textContainer"
  >
    <h3
      className="name"
    >
      gift
    </h3>
    <h4
      className="title"
    >
      Price: Â£
      20
    </h4>
    <h4
      className="title"
    >
      Requested: 
      5
    </h4>
    <h4
      className="title"
    >
      Remaining: 
      10
    </h4>
  </div>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltipPosition="bottom-center"
    touch={true}
  >
    <ImageEdit />
  </IconButton>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltipPosition="bottom-center"
    touch={true}
  >
    <ActionDelete />
  </IconButton>
</div>
`;
